Libraries/PeriphDrivers/Include/MAX32657/i3c.h

```diff
@@ -0,0 +1,621 @@
+    MXC_I3C_IBI_TYPE_CONTROLLER_REQ = MXC_V_I3C_CONT_STATUS_IBITYPE_CONT_REQ, ///<
+    MXC_I3C_IBI_TYPE_HOTJOIN_REQ = MXC_V_I3C_CONT_STATUS_IBITYPE_HOTJOIN_REQ, ///<
+} mxc_i3c_ibi_type_t;
+
+/* CCC request flags */
+#define MXC_I3C_CCC_HAS_DEFINING_BYTE (1U << 0)
+#define MXC_I3C_CCC_HAS_SUB_COMMAND (1U << 1)
+#define MXC_I3C_CCC_HAS_DATA (1U << 2)
+
+/**
+ * @brief Common Command Code request structure.
+ *
+ * Broadcast and direct commands are distinguished by MSB, i.e. if MSB being set
+ * implies a direct CCC while MSB being 0 implies a broadcast CCC.
+ *
+ */
+struct _i3c_ccc_req_t {
+    uint8_t ccc; ///< CCC command to send.
+    uint8_t target_addr : 7; ///< Target address if CCC is a direct command. Ignored
+    ///< if CCC is a broadcast command.
+    uint8_t read_write : 1; ///< Set to 1 for read commands and set to 0 for write
+    ///< commands. Ignored for broadcast CCCs.
+    uint8_t flags; ///< See request flags above.
+    uint8_t def_byte; ///< Optional defining byte. Defined by CCC.
+    uint8_t subCmd; ///< Optional sub-command. Defined by CCC. Only used by direct
+    ///< CCCs.
+    unsigned char *tx_buf; ///< Optional data bytes to send.
+    uint8_t tx_len; ///< Length of optional data.
+    unsigned char *rx_buf; ///< Optional data bytes to read.
+    uint8_t rx_len; ///< Length of optional data.
+};
+
+/**
+ * @brief Private SDR request structure.
+ *
+ * SDR read and write to an I3C target.
+ *
+ */
+struct _i3c_req_t {
+    uint8_t target_addr : 7; ///< Target address.
+    uint8_t is_i2c : 1; ///< If this is a legacy I2C transfer.
+    uint8_t stop; ///< Send a STOP after the transaction.
+    unsigned char *tx_buf; ///< Optional data bytes to send.
+    uint16_t tx_len; ///< Length of optional data.
+    unsigned char *rx_buf; ///< Optional data bytes to read.
+    uint16_t rx_len; ///< Length of optional data.
+};
+
+/**
+ * @brief   IBI payload request callback. Write additional byte to \a byte.
+ *
+ * This function will be called as long as non-zero is returned.
+ *
+ * @return  Non-zero if a byte is written to \a byte, 0 to indicate no more additional
+ * bytes left to send.
+ */
+typedef int (*mxc_i3c_ibi_getbyte_t)(mxc_i3c_regs_t *i3c, unsigned char *byte);
+
+/***** Function Prototypes *****/
+
+/* ************************************************************************* */
+/* Control/Configuration functions                                           */
+/* ************************************************************************* */
+
+/**
+ * @brief   Initialize and enable I3C peripheral.
+ *
+ * @note    On default this function enables I3C peripheral clock and I3C gpio pins.
+ *          If you wish to skip clock and gpio initialization, define MSDK_NO_GPIO_CLK_INIT
+ *          flag in project.mk file.
+ *
+ * @param   i3c         Pointer to I3C registers (selects the I3C block used).
+ * @param   targetMode  Whether to put the device in controller or target mode. Use
```

Comment:
Update targetMode type to `bool` from `<stdbool.h>`

Reply:
Done.